node/src/accountant/scanners/mod.rs

```diff
@@ -894,10 +897,75 @@ impl ReceivableScanner {
             earning_wallet,
             receivable_dao,
             banned_dao,
+            persistent_configuration,
             financial_statistics,
         }
     }
 
+    fn update_start_block_while_no_received_transactions_found(
+        &mut self,
+        new_start_block: u64,
+        logger: &Logger,
+    ) {
+        info!(
+            logger,
+            "No new received payments were detected during the scanning process."
+        );
+        match self
+            .persistent_configuration
+            .set_start_block(new_start_block)
+        {
+            Ok(()) => debug!(logger, "Start block updated to {}", new_start_block),
+            Err(e) => panic!(
```

Comment:
I think it should be a warning or error log instead of a panic. This may lead the Node to die even though the condition is not fatal.

Reply:
I need to oppose to you.  It's been a long-living rule we've agreed on concerning the database: if it fails at any place where an error is not expected to be a legitimate option we should immediately kill the Node - because we don't know what such a bad thing happened with the database - preventing even further damage of the data. You can ask Dan.